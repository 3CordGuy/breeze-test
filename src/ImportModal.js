import React from "react";
import { Button, Modal } from "semantic-ui-react";

const ImportModal = ({
    showModal,
    onClose,
    onConfirm,
    closeModal,
    confirmText,
    children,
}) => {
    return (
        <div>
            <Modal
                open={showModal}
                closeOnEscape={false}
                closeOnDimmerClick={false}
                onClose={onClose}
            >
                <Modal.Header>We are almost ready to import...</Modal.Header>
                <Modal.Content>{children}</Modal.Content>
                <Modal.Actions>
                    <Button onClick={closeModal} basic negative>
                        Cancel
                    </Button>
                    <Button
                        onClick={onConfirm}
                        positive
                        labelPosition="right"
                        icon="checkmark"
                        content={confirmText || "Yes"}
                    />
                </Modal.Actions>
            </Modal>
        </div>
    );
};

export default ImportModal;
