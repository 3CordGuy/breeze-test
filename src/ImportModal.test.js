import React from "react";
import { shallow } from "enzyme";
import ImportModal from "./ImportModal";

let wrapper;

describe("<ImportModal />", () => {
    beforeAll(() => {
        wrapper = shallow(<ImportModal />);

        wrapper.setProps({ showModal: true });
    });

    test("should match the snapshot", () => {
        expect(wrapper).toMatchSnapshot();
    });
});
