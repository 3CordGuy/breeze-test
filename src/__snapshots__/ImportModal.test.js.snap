// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImportModal /> should match the snapshot 1`] = `
<div>
  <Modal
    centered={true}
    closeOnDimmerClick={false}
    closeOnDocumentClick={false}
    closeOnEscape={false}
    dimmer={true}
    eventPool="Modal"
    open={true}
  >
    <ModalHeader>
      We are almost ready to import...
    </ModalHeader>
    <ModalContent />
    <ModalActions>
      <Button
        as="button"
        basic={true}
        negative={true}
      >
        Cancel
      </Button>
      <Button
        as="button"
        content="Yes"
        icon="checkmark"
        labelPosition="right"
        positive={true}
      />
    </ModalActions>
  </Modal>
</div>
`;
